storage_controller/src/persistence.rs

```diff
@@ -482,9 +505,9 @@ impl Persistence {
         .await
     }
 
-    pub(crate) async fn hard_delete_node(&self, del_node_id: NodeId) -> DatabaseResult<()> {
+    pub(crate) async fn delete_node(&self, del_node_id: NodeId) -> DatabaseResult<()> {
         use crate::schema::nodes::dsl::*;
-        self.with_measured_conn(DatabaseOperation::HardDeleteNode, move |conn| {
+        self.with_measured_conn(DatabaseOperation::DeleteNode, move |conn| {
             Box::pin(async move {
                 // You can hard delete a node if it was soft deleted before.
```

Comment:
nit: This comment should be updated: You can delete a node if it has a tombstone on it